### DISCUSSION TITLE
Add --if-present flag to turbo run

### DISCUSSION THREAD

**Discussion by zanona** [vercel/turborepo#1106] [Ideas]
I guess this falls in line with `npm run -ws --if-preset <command>`, because at the moment if I define a task on `turbo.json` as below:

turbo.json
```json
{
   "pipeline": {
     "dev": { "dependsOn": ["build"] }
   }
}
```

package.json
```json
{
   "name": "@scope/a",
   "scripts": {
     "build": "sleep 10"
   }
}
```

And then call `turbo run dev`, even though `@scope/a` doesn't have a script called `dev`, it will still run its build step and delay the process 10 seconds unnecessarily here.
So the idea would be that, since that the build script is not dependent, nor necessary for the execution of the `dev` task, we could either make that a default and skip unnecessary tasks or allow `turbo run dev --if-present` to be defined.
Hopefully I am not overseeing anything vital here :smiley:

**Comment by michaelkplai** [comment]
May be related to #937.

**Reply by zanona** [reply]
Thanks for the heads up
